recipes/mysql-concpp/all/conanfile.py

```diff
@@ -0,0 +1,207 @@
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def _package_folder_dep(self, dep):
+        return self.dependencies[dep].package_folder.replace("\\", "/")
+
+    def _include_folder_dep(self, dep):
+        return self.dependencies[dep].cpp_info.includedirs[0].replace("\\", "/")
+
+    def _lib_folder_dep(self, dep):
+        return self.dependencies[dep].cpp_info.libdirs[0].replace("\\", "/")
+
+    def generate(self):
+
+        tc = CMakeToolchain(self, generator="Ninja") if not is_msvc(self) else CMakeToolchain(self)
+
+        # OpenSSL
+        tc.cache_variables["WITH_SSL"] = "SYSTEM"
+        tc.cache_variables["BUNDLE_DEPENDENCIES"] = "OFF"
+        # LZ4 patches
+        tc.cache_variables["WITH_LZ4"] = "TRUE"
+        # ZLIB patches
+        tc.cache_variables["WITH_ZLIB"] = "SYSTEM"
+        tc.cache_variables["ZLIB_DIR"] = self._package_folder_dep("zlib")
+        # ZSTD patches
+        tc.cache_variables["WITH_ZSTD"] = "TRUE"
+        # Build patches
+        tc.cache_variables["BUILD_STATIC"] = not self.options.shared
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+
+        # Windows patches
+        if self.settings.os == "Windows":
+            # OpenSSL patches
+            tc.cache_variables["SSL_DIR"] = self._package_folder_dep("openssl")
+            if is_msvc(self):
+                # Boost patches
+                tc.cache_variables["BOOST_DIR"] = self._package_folder_dep("boost")
+
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        # Alias for the OpenSSL target
+        deps.set_property("openssl", "cmake_target_name", "mysql::openssl")
+        deps.generate()
+
+    def _patch_sources(self):
+        if not self.options.shared and is_msvc(self):
+            replace_in_file(self, os.path.join(self.source_folder, "install_layout.cmake"),
+                                "set(LIB_NAME_STATIC \"${LIB_NAME}-mt\")",
+                                "set(LIB_NAME_STATIC \"${LIB_NAME_STATIC}-mt\")",
+                                strict=False)
+
+        # ZSTD patch
+        replace_in_file(self, os.path.join(self.source_folder, "cdk", "extra", "zstd", "CMakeLists.txt"),
+                            "enable_pic()",
+                            "enable_pic()\n"\
+                            "add_compile_definitions(ZSTD_DISABLE_ASM)",
+                            strict=False)
+
+        # Apple patches
+        if is_apple_os(self) and cross_building(self):
+            print(f"Building for {str(self.settings.arch)}")
+            patch = f"set(CMAKE_OSX_ARCHITECTURES \"{str(self.settings.arch)}\" CACHE INTERNAL \"\" FORCE)\n"
```

Comment:
```suggestion
            patch = f"set(CMAKE_OSX_ARCHITECTURES \"{self.settings.arch}\" CACHE INTERNAL \"\" FORCE)\n"
```
Should use `self.output.info()` instead of `print()` in recipes, but that was not adding much anyway.